rest/api_test.go

```diff
@@ -3137,6 +3137,36 @@ func TestDocSyncFunctionExpiry(t *testing.T) {
 	log.Printf("value: %v", value)
 }
 
+func TestCustomCookieName(t *testing.T) {
+
+	var rt RestTester
+	defer rt.Close()
+
+	customCookieName := "TestCustomCookieName"
+	rt.DatabaseConfig = &DbConfig{
+		Name:              "db",
+		SessionCookieName: customCookieName,
+	}
+
+	response := rt.SendAdminRequest("POST", "/db/_user/", `{"name":"user1", "password":"1234"}`)
+	assertStatus(t, response, 201)
+
+	resp := rt.SendAdminRequest("POST", "/db/_session", `{"name": "user1"}`)
+	assert.Equals(t, resp.Code, 200)
+
+	sessionResponse := struct {
+		SessionID  string `json:"session_id"`
+		Expires    string `json:"expires"`
+		CookieName string `json:"cookie_name"`
+	}{}
+
+	err := json.Unmarshal(resp.Body.Bytes(), &sessionResponse)
+	assertNoError(t, err, "Unexpected error")
+
+	assert.Equals(t, sessionResponse.CookieName, customCookieName)
+
```

Comment:
This doesn't test authentication with the custom cookie name, just the cookie name in the response body.  Were you able to (even manually) validate (a) the name being set in the Set-Cookie header, and (b) using the cookie to authenticate?

Reply:
Nope, I haven't done that level of validation, but I will look into adding it

Reply:
@adamcfraser ok I pushed up a new change that adds this testing functionality.

Not realizing I needed to set this flag on the resttester really tripped me up:

```
	rt.noAdminParty = true
```

Without that, I wasn't able to test authentication since it disabled the auth checking.  Only managed to find that flag through a fair amount of debugging.